tests/kafkatest/tests/core/log_dir_failure_test.py

```diff
@@ -0,0 +1,179 @@
+                   "Leader node %d should be in isr set %s" % (broker_idx, str(self.kafka.isr_idx_list(self.topic1)))
+
+            self.logger.debug("Making log dir %s inaccessible" % (KafkaService.DATA_LOG_DIR_1))
+            cmd = "sudo chmod a-rw %s -R" % (KafkaService.DATA_LOG_DIR_1)
+            broker_node.account.ssh(cmd, allow_fail=False)
+
+            if bounce_broker:
+                self.kafka.restart_node(broker_node, clean_shutdown=True)
+
+            # Verify the following:
+            # 1) The broker with offline log directory is not the leader of the partition of topic1
+            # 2) The broker with offline log directory is not in the ISR
+            # 3) The broker with offline log directory is still online
+            # 4) Messages can still be produced and consumed from topic1
+            wait_until(lambda: self.kafka.leader(self.topic1, partition=0) != broker_node,
+                       timeout_sec=60,
+                       err_msg="Broker %d should not be leader of topic %s and partition 0" % (broker_idx, self.topic1))
+            assert self.kafka.alive(broker_node), "Broker %d should be still online" % (broker_idx)
+            wait_until(lambda: broker_idx not in self.kafka.isr_idx_list(self.topic1),
+                       timeout_sec=60,
+                       err_msg="Broker %d should not be in isr set %s" % (broker_idx, str(self.kafka.isr_idx_list(self.topic1))))
+
+            self.stop_producer_and_consumer()
+            self.validate()
+
+            # Shutdown all other brokers so that the broker with offline log dir is the only online broker
+            offline_nodes = []
+            for node in self.kafka.nodes:
+                if broker_node != node:
+                    offline_nodes.append(node)
+                    self.logger.debug("Hard shutdown broker %d" % (self.kafka.idx(node)))
+                    self.kafka.signal_node(node, sig=signal.SIGKILL)
+
+            # Verify the following:
+            # 1) The broker with offline directory is the only in-sync broker of the partition of topic2
+            # 2) Messages can still be produced and consumed from topic2
+            self.producer = VerifiableProducer(self.test_context, self.num_producers, self.kafka, self.topic2,
+                                               throughput=self.producer_throughput, offline_nodes=offline_nodes)
+            self.consumer = ConsoleConsumer(self.test_context, self.num_consumers, self.kafka, self.topic2, group_id="test-consumer-group-2",
+                                            new_consumer=False, consumer_timeout_ms=60000, message_validator=is_int)
+            self.start_producer_and_consumer()
+
+            assert self.kafka.isr_idx_list(self.topic2) == [broker_idx], \
+                   "In-sync replicas of topic %s and partition 0 should be %s" % (self.topic2, str([broker_idx]))
+
+            self.stop_producer_and_consumer()
+            self.validate()
```

Comment:
Is the test from line 161 to 174 necessary? It doesn't seem to be directly related to offline disks.